### ISSUE TITLE
[Feature Request] Ability to customise whitespace for individual statements.

### ISSUE THREAD

**Issue by Ovyerus** [palantir/tslint#4075]
### Feature request

**Is your feature request that we implement a new rule?**
Not really (could be done like that though).

**Is your feature request related to a problem? Please describe.**
My coding style is somewhat specific to what I prefer, and that ends up including some weird stuff like my whitespace for `if ()` and `catch()` being different despite both being statements. This isn't an issue with ESLint as I'm able to customise the whitespace on a per-keyword statement basis, however I'm unable to do this with TSLint.

**Describe the solution you'd like**
An extension to the `whitespace` rule (or a separate rule) that allows you to specify exactly how you want your whitespace, similar to ESLint's [keyword-spacing](https://eslint.org/docs/rules/keyword-spacing) rule.

An example implementation could be something like this in `tslint.json`:
```json
"whitespace": [
    true,
    "check-decl",
    ...,
    {
        "check-branch": {
            "if": {"before": true, "after": true},
            "catch": {"before": true, "after": false"}
        }
    }
]
```
Which would then let the following code be valid:
```ts
if (true) {
    try {
        throw new Error('foo');
    } catch(err) {
        console.error(err);
    }
}
```

**Comment by JoshuaKGoldberg** [comment]
Seems like this is somewhat a duplicate of #628 and all its related issues. Closing for housekeeping to keep our requested issues list small; please see discussions there.
